packages/databricks-vscode/src/feature-manager/FeatureManager.ts

```diff
@@ -0,0 +1,105 @@
+import {Event, EventEmitter, Disposable} from "vscode";
+import {DisabledFeature} from "./DisabledFeature";
+
+export type FeatureEnableAction = () => Promise<void>;
+export interface FeatureState {
+    avaliable: boolean;
+    reason?: string;
+    action?: FeatureEnableAction;
+}
+
+export type FeatureId = "debugging.dbconnect";
+export const disabledFeatures: FeatureId[] = ["debugging.dbconnect"];
```

Comment:
I'd rather have `disabledFeatures` a constructor argument and pass it in from `extension.ts`